master/buildbot/status/worker.py

```diff
@@ -41,6 +42,24 @@ def __init__(self, name):
         self.graceful_callbacks = []
         self.pause_callbacks = []
         self.connect_times = []
+        self.info = Properties()
+
+    def setInfo(self, info):
+        self.admin = bytes2unicode(info.get("admin"))
+        self.host = bytes2unicode(info.get("host"))
+        self.access_uri = info.get("access_uri", None)
+        if self.access_uri:
+            self.access_uri = bytes2unicode(self.access_uri)
+        self.version = bytes2unicode(info.get("version", b"(unknown)"))
+
+        # store everything as Properties
+        for k, v in info.items():
+            if k in ('environ', 'worker_commands'):
+                continue
+            self.info.setProperty(k, v, "Worker")
```

Comment:
Why storing as a Property, why not just store the info dict as is?

Reply:
I am 100% convinced that the master side will only get bytes and not string no matter how much I decode on worker.

Reply:
I tried with dict and it did not work well. I can try again if it is important.